[Failing test] `Create and reconcile ManagedSeed` test is failing

**How to categorize this issue?**
/area testing
/kind bug
/priority 3

**What happened**:

The last 9 runs of the test fail with identical errors:

![Screenshot 2021-05-13 at 18 15 03](https://user-images.githubusercontent.com/9372594/118146666-809dba00-b417-11eb-84f8-0962f546c16f.png)

Example logs of a failed run:

<details>

<summary>Test run logs</summary>

```
FAIL
FAIL  github.com/gardener/gardener/test/system/managed_seed_creation  3630.293s
FAIL
--- FAIL: TestManagedSeedCreation (3630.24s)
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
Ran 1 of 1 Specs in 3630.237 seconds

/src/test/system/managed_seed_creation/create_test.go:46
[Fail] ManagedSeed creation testing [It] [MANAGEDSEED] Create and reconcile ManagedSeed 

Summarizing 1 Failure:


------------------------------
  /src/test/system/managed_seed_creation/create_test.go:46

  occurred
      retry failed with context deadline exceeded, last error: seeds.core.gardener.cloud "tm96c-pz9" not found
      }
          },
              },
                  Code: 404,
                  },
                      RetryAfterSeconds: 0,
                      Causes: nil,
                      UID: "",
                      Kind: "seeds",
                      Group: "core.gardener.cloud",
                      Name: "tm96c-pz9",
                  Details: {
                  Reason: "NotFound",
                  Message: "seeds.core.gardener.cloud \"tm96c-pz9\" not found",
                  Status: "Failure",
                  },
                      RemainingItemCount: nil,
                      Continue: "",
                      ResourceVersion: "",
                      SelfLink: "",
                  ListMeta: {
                  TypeMeta: {Kind: "", APIVersion: ""},
              ErrStatus: {
          err: {
          ctxError: {},
      <*retry.Error | 0xc0006340e0>: {
  Unexpected error:

  /src/test/framework/gingko_utils.go:26
  [MANAGEDSEED] Create and reconcile ManagedSeed [It]
/src/test/system/managed_seed_creation/create_test.go:34
ManagedSeed creation testing
• Failure [3630.237 seconds]

time="2021-05-13T13:51:28Z" level=info msg="[GARDENER]"
  /src/test/framework/gingko_utils.go:41
[AfterEach] ManagedSeed creation testing
time="2021-05-13T13:51:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:50:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:50:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:49:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:49:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:48:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:48:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:47:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:47:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:46:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:46:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:45:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:45:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:44:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:44:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:43:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:43:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:42:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:42:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:41:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:41:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:40:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:40:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:39:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:39:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:38:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:38:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:37:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:37:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:36:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:36:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:35:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:35:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:34:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:34:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:33:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:33:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:32:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:32:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:31:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:31:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:30:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:30:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:29:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:29:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:28:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:28:00Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:27:30Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:26:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:26:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:25:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:25:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:24:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:24:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:23:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:23:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:22:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:22:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:21:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:21:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:20:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:20:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:19:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:19:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:18:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:18:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:17:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:17:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:16:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:16:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:15:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:15:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:14:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:14:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:13:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:13:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:12:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:12:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:11:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:11:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:10:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:10:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:09:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:09:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:08:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:08:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:07:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:07:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:06:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:06:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:05:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:05:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:04:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:04:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:03:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:03:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:02:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:02:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:01:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:01:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:00:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T13:00:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:59:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:59:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:58:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:58:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:57:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:57:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:56:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:56:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:55:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:55:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:54:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:54:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:53:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:53:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:52:59Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:52:29Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:51:58Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:51:28Z" level=info msg="Error while waiting for seed to be created: seeds.core.gardener.cloud \"tm96c-pz9\" not found"
time="2021-05-13T12:51:28Z" level=info msg="Waiting until seed tm96c-pz9 is successfully reconciled"
status: {}
    name: tm96c-pz9
  shoot:
            value: dev-cp-mig2sfxv
          - key: test.gardener.cloud/test-run
          taints:
              visible: false
            scheduling:
          settings:
            namespace: garden
            name: seed-tm96c-pz9
          secretRef:
            type: ""
            region: ""
          provider:
            services: ""
            pods: ""
          networks:
          dns: {}
            secretRef: {}
            provider: azure
          backup:
        spec:
          creationTimestamp: null
        metadata:
      seedConfig:
      kind: GardenletConfiguration
      apiVersion: gardenlet.config.gardener.cloud/v1alpha1
    config:
  gardenlet:
spec:
  namespace: garden
  name: tm96c-pz9
  creationTimestamp: null
metadata:
time="2021-05-13T12:50:58Z" level=info msg="Creating managed seed tm96c-pz9"
time="2021-05-13T12:50:58Z" level=info msg="Shoot tm96c-pz9 was successfully updated!"
  /src/test/framework/gingko_utils.go:26
[It] [MANAGEDSEED] Create and reconcile ManagedSeed
  /src/test/framework/gingko_utils.go:51
[BeforeEach] ManagedSeed creation testing
  /src/test/framework/gingko_utils.go:26
  [MANAGEDSEED] Create and reconcile ManagedSeed
ManagedSeed creation testing 

Will run 1 of 1 specs
Random Seed: 1620910258
==============================================
Running Suite: ManagedSeed Creation Test Suite
=== RUN   TestManagedSeedCreation
```

</details>


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:


- Gardener version: v1.23.0-dev-b48ae95e29853f7bb590c6838b4e2aa70c6c98d2
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/charts/gardener/controlplane/charts/application/templates/clusterrole-seed-bootstrapper.yaml
+++ b/charts/gardener/controlplane/charts/application/templates/clusterrole-seed-bootstrapper.yaml
@@ -11,3 +11,9 @@ rules:
   verbs:
   - create
   - get
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - certificatesigningrequests/seedclient
+  verbs:
+  - create

--- a/docs/deployment/deploy_gardenlet_manually.md
+++ b/docs/deployment/deploy_gardenlet_manually.md
@@ -213,6 +213,12 @@ rules:
     verbs:
       - create
       - get
+  - apiGroups:
+      - certificates.k8s.io
+    resources:
+      - certificatesigningrequests/seedclient
+    verbs:
+      - create
 ---
 # A kubelet/gardenlet authenticating using bootstrap tokens is authenticated as a user in the group system:bootstrappers
 # Allows the Gardenlet to create a CSR

